### PR TITLE
Revert "remove RunAPIAnalysisBuilderAsJob option from perferences UI"

### PR DESCRIPTION
**Pull request by jukzi**
Reverts eclipse-pde/eclipse.pde#568

reason: Parallel API Analysis not stable, see

https://github.com/eclipse-pde/eclipse.pde/issues/1107
https://github.com/eclipse-platform/eclipse.platform/issues/200

### CODE REVIEW

**Comment by iloveeclipse** [comment]
Change in PDEUIMessages is missing.

**Comment by gireeshpunathil** [comment]
If parallel analysis is not stable, why are we introducing this option to the user again, instead of disallowing parallel analysis itself?

**Comment by iloveeclipse** [comment]
Because it is hard to make it stable and user should be able to choose what way should Eclipse work - slow and stable or fast and unstable.

**Comment by laeubi** [comment]
> slow and stable or fast and unstable.

We might rename the option into "fail fast mode" :-D

**Comment by gireeshpunathil** [comment]
> Because it is hard to make it stable and user should be able to choose what way should Eclipse work - slow and stable or fast and unstable.

along with the revert, are we warning the user around the checkbox saying: `on: fast and unstable`, `off: slow but stable` ? if not, this revert alone looks to me like misguiding the user!

**Comment by iloveeclipse** [comment]
There is not much to warn, every code has bugs, so should every preference has a warning?

**Comment by laeubi** [comment]
> There is not much to warn, every code has bugs, so should every preference have a warning?

It should warn if there are unforeseen consequence. e.g as A user I might need to know what will happen if I choose the option e.g. what does it means to run it as a job and if it is faster, why I should NOT choose that option ...

Given that here
- https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/898

it was even asked for explaining the whole story of JUnit4/5 in a single checkbox by the same audience of people it seems fair to address @gireeshpunathil concerns here in the same way...

**Comment by jukzi** [comment]
i vote for: disable parallel by default, add the UI option to enable again.
Reasoning: the few people who ever will find time to enable the option will know that they enabled something non-standard. it feels like common sense that something that is meant to better but not enabled by default will come with some drawback.

**Comment by merks** [comment]
BTW, I provisioned a new IDE today with the latest of everything, and it had an API error that went away with a clean build; very annoying.

**Comment by gireeshpunathil** [comment]
> There is not much to warn, every code has bugs, so should every preference have a warning?

no, but if I know a feature has a bug in it, then I would definitely warn, if bringing it back and present to the user.

**Comment by iloveeclipse** [comment]
> i vote for: disable parallel by default, add the UI option to enable again.

That would be my prefered solution.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/pderesources.properties [hunk 1]
```diff
@@ -368,6 +368,7 @@ MainPreferencePage_ShowTargetStatus=S&how current target platform in status bar
 MainPreferencePage_updateStale=&Update stale manifest files prior to launching
 MainPreferencePage_WorkspacePluginsOverrideTarget=Workspace p&lug-ins override target platform plug-ins with the same id
 MainPreferencePage_DisableAPIAnalysisBuilder=&Disable API analysis builder
+MainPreferencePage_RunAPIAnalysisBuilderAsJob=&Run API analysis parallel to the build job
```
Comment (laeubi) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/pderesources.properties:371]:
```suggestion
MainPreferencePage_RunAPIAnalysisBuilderAsJob=&Run API analysis in the background (experimental)
```

**Comment by jukzi** [comment]
> it blocks all user operation

that could be solved like in JDT, where the build jobs checks for interrupt/cancel

**Review by gireeshpunathil — APPROVED** [review]

**Pull request merged**